app/src/main/res/layout/activity_bill.xml

```diff
@@ -255,32 +268,69 @@
                             android:layout_height="wrap_content"
-                            android:layout_marginTop="@dimen/dp_2"
-                            android:onClick="@{()-> viewModel.onAddBillClick()}"
-                            app:srcCompat="@drawable/ic_cart_60dp"
-                            tools:ignore="ContentDescription"/>
+                            bind:errorText="@{viewModel.formError}">
+
+                            <android.support.v7.widget.CardView
+                                android:layout_width="wrap_content"
+                                android:layout_height="wrap_content"
+                                android:paddingRight="@dimen/dp_10"
+                                app:cardBackgroundColor="@color/colorPrimary"
+                                app:cardCornerRadius="@dimen/dp_5"
+                                car_view:cardElevation="@dimen/dp_2"
+                                >
+
+                                <ImageView
+                                    android:layout_width="wrap_content"
+                                    android:layout_height="wrap_content"
+                                    android:layout_marginTop="@dimen/dp_2"
+                                    android:onClick="@{()-> viewModel.onAddBillClick()}"
+                                    android:padding="@dimen/dp_3"
+                                    app:srcCompat="@drawable/ic_cart_60dp"
+                                    tools:ignore="ContentDescription"/>
+                            </android.support.v7.widget.CardView>
+                        </android.support.design.widget.TextInputLayout>
                     </LinearLayout>
                 </LinearLayout>
             </android.support.v7.widget.CardView>
 
             <android.support.v7.widget.RecyclerView
                 android:layout_width="match_parent"
-                android:layout_height="wrap_content"
+                android:layout_height="0dp"
+                android:layout_weight="1"
+                android:padding="@dimen/dp_5"
                 bind:adapter="@{viewModel.adapter}"
                 bind:layoutManager="@{LayoutManager.linear()}"/>
 
-            <android.support.v7.widget.AppCompatButton
+            <LinearLayout
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:layout_marginBottom="@dimen/dp_10"
-                android:layout_marginTop="@dimen/dp_10"
-                android:background="@color/colorPrimary"
-                android:padding="@dimen/dp_12"
-                android:text="@string/title_create"
-                android:textColor="@android:color/white"/>
+                android:orientation="vertical">
+
+                <TextView
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="right"
```

Comment:
WARNING: [RtlHardcoded] Use "`end`" instead of "`right`" to ensure correct behavior in right-to-left locales
 Using `Gravity#LEFT` and `Gravity#RIGHT` can lead to problems when a layout is rendered in locales where text flows from right to left. Use `Gravity#START` and `Gravity#END` instead. Similarly, in XML `gravity` and `layout_gravity` attributes, use `start` rather than `left`.

For XML attributes such as paddingLeft and `layout_marginLeft`, use `paddingStart` and `layout_marginStart`. *NOTE*: If your `minSdkVersion` is less than 17, you should add *both* the older left/right attributes *as well as* the new start/right attributes. On older platforms, where RTL is not supported and the start/right attributes are unknown and therefore ignored, you need the older left/right attributes. There is a separate lint check which catches that type of error.

(Note: For `Gravity#LEFT` and `Gravity#START`, you can use these constants even when targeting older platforms, because the `start` bitmask is a superset of the `left` bitmask. Therefore, you can use `gravity="start"` rather than `gravity="left|start"`.)